core/includes/form.inc

```diff
@@ -3716,8 +3724,17 @@ function theme_details($variables) {
     $element['#attributes']['open'] = 'open';
   }
 
+  $summary_content = '<span>' . $element['#summary'] . '</span>';
+
+  // Custom labels for the open/close toggle.
+  $more_less_toggle = '';
+  if (isset($element['#open_label']) && isset($element['#close_label'])) {
+    $more_less_toggle = '<span class="more-less-toggle-label" data-open-label="' . $element['#open_label'] . '" data-close-label="' . $element['#close_label'] . '"></span>';
+    $element['#attributes']['class'][] = 'more-less-toggle';
+  }
+
   $output = '<details' . backdrop_attributes($element['#attributes']) . '>';
-  $output .= '<summary><span>' . $element['#summary'] . '</span></summary>';
+  $output .= '<summary>' . $summary_content . $more_less_toggle . '</summary>';
```

Comment:
```suggestion
  if (empty($more_less_toggle)) {
    $output .= '<summary>';
  }
  else {
    $output .= '<summary class="more-less-toggle">';
  }
  $output .= $summary_content . $more_less_toggle . '</summary>';
```

Reply:
Just realized that the appending of `$more_less_toggle` and closing tag might as well go inside the `if` and `else`.

Reply:
I initially though to do something like this instead:
```suggestion
  $summary_class = !empty($more_less_toggle) ? ' class="more-less-toggle"' : '';
  $output .= '<summary' . $summary_class . '>' . $summary_content . $more_less_toggle . '</summary>';
```

I then thought that it would in general be useful to be able to specify attributes for the `<summary>` element besides the `<details>` element (which is done via the `#attributes` property), so I've introduced a `'#summary_attributes'` property for that, and set the `more-less-toggle` class that way.

Reply:
... @bugfolder since the `<summary>` element is always nested within the `<details>` element, and the `more-less-toggle` class is already being added to the `<details>` element, we can target the summary in CSS via `details.more-less-toggle summary` (or just `.more-less-toggle summary`). Do you still think that we really need to be repeating the same class in both elements?

Reply:
No, as long as it's targetable, that's all we need.